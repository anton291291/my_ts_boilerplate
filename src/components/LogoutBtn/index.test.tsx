import React from 'react';
import { Router } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import { LogoutBtn } from '.';

describe('LogoutBtn', () => {
    const mockPush = jest.fn();

    const historyMock = { push: mockPush, location: {}, listen: jest.fn() };

    const output = mount(
        <Router history={historyMock}>
            <LogoutBtn />
        </Router>
    );

    test('should render correctly', () => {
        expect(toJson(output)).toMatchSnapshot();
    });

    test('handelLogout func should delete from localStorage', () => {
        output.simulate('click');
        expect(localStorage.removeItem).toHaveBeenCalledWith('name');
    });

    test('handelLogout should push the history', () => {
        expect(mockPush).toHaveBeenCalledWith('/');
    });
});
