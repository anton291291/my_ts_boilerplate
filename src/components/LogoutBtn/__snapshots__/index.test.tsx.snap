// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutBtn should render correctly 1`] = `
<Router
  history={
    Object {
      "listen": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "location": Object {},
      "push": [MockFunction],
    }
  }
>
  <LogoutBtn>
    <Logout
      onClick={[Function]}
    >
      <ForwardRef
        className="Logout-sc-16zpe0z-0 dLiqNe"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(SvgIcon))
          className="Logout-sc-16zpe0z-0 dLiqNe"
          onClick={[Function]}
        >
          <ForwardRef(SvgIcon)
            className="Logout-sc-16zpe0z-0 dLiqNe"
            classes={
              Object {
                "colorAction": "MuiSvgIcon-colorAction",
                "colorDisabled": "MuiSvgIcon-colorDisabled",
                "colorError": "MuiSvgIcon-colorError",
                "colorPrimary": "MuiSvgIcon-colorPrimary",
                "colorSecondary": "MuiSvgIcon-colorSecondary",
                "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                "root": "MuiSvgIcon-root",
              }
            }
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root Logout-sc-16zpe0z-0 dLiqNe"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </ForwardRef(SvgIcon)>
        </WithStyles(ForwardRef(SvgIcon))>
      </ForwardRef>
    </Logout>
  </LogoutBtn>
</Router>
`;
